File: CCC/2017 Senior CCC/S1_Sum_Game.py
N = int(input())
team1 = list(map(int,input().split()))
team2 = list(map(int,input().split()))
for i in range(1,N):#since both lists have same length, loop through both and create a prefix sum array
  team1[i]+=team1[i-1]
  team2[i]+=team2[i-1]
  
while N:#if any of the two lists are equal at a given index print that day
  if team1[N-1] == team2[N-1]:
    print(N)
    break
  N-=1
if N==0:#if N reaches that means it will never be equal on a given day; thus, print 0 = N.
  print(N)

File: CCC/2017 Senior CCC/S1_Sum_Game.cpp
#include <iostream>

using namespace std;
int N;
int team1[1000000]={0};
int team2[1000000]={0};
bool notPrinted = true;
int main()
{

    cin >> N;

    for (int i = 0; i<N; i++){
        cin >> team1[i];
    }
    for (int i = 0; i<N; i++){
        cin >> team2[i];
    }

    for (int i = 1; i<N;i++){ //make the prefix array for both teams
        team1[i]+=team1[i-1];
        team2[i]+=team2[i-1];
    }
    for(int i = N-1; i>0; i--){//check if at any date the two are equal and if so print it
        if (team1[i]==team2[i]){
            cout << i << endl;
            notPrinted = false;
        }
    }
    if(notPrinted){//if nothing was printed that means it was equal on no date meaning print 0
        cout <<0<<endl;
    }
    return 0;
}
